testrepository/tests/commands/test_run.py

```diff
@@ -539,3 +622,90 @@ def test_returncode_nonzero_fail_appended_to_content(self):
             expected_content = _b('foo\nbar\ntest: process-returncode\n'
                 'failure: process-returncode [\n returncode 1\n]\n')
         self.assertEqual(expected_content, content)
+
+
+class TestLoadingIds(BaseTestCommand):
+
+    scenarios = [
+        ('noprofile', {'profile': ''}),
+        ('profile', {'profile': 'DEFAULT'})]
+
+    def test_load_list_passes_ids(self):
+        list_file = tempfile.NamedTemporaryFile()
+        self.addCleanup(list_file.close)
+        expected_ids = set(['foo', 'quux', 'bar'])
+        load_ids = set(expected_ids)
+        if self.profile:
+            load_ids = apply_profiles([self.profile], load_ids)
+        expected_ids = apply_profiles(['DEFAULT'], expected_ids)
+        write_list(list_file, load_ids)
+        list_file.flush()
+        ui, cmd = self.get_test_ui_and_cmd(
+            options=[('load_list', list_file.name)])
+        cmd.repository_factory = memory.RepositoryFactory()
+        self.setup_repo(cmd, ui)
+        self.set_config(
+            '[DEFAULT]\ntest_command=foo $IDOPTION\ntest_id_option=--load-list $IDFILE\n')
+        params, capture_ids = self.capture_ids()
+        self.useFixture(MonkeyPatch(
+            'testrepository.testcommand.TestCommand.get_run_command',
+            capture_ids))
+        cmd_result = cmd.execute()
+        self.assertEqual([
+            ('results', Wildcard),
+            ('summary', True, 0, -3, None, None, [('id', 1, None)])
+            ], ui.outputs)
+        self.assertEqual(0, cmd_result)
+        self.assertEqual([[Wildcard, expected_ids, [], None]], params)
```

Comment:
This and other tests are fairly lengthy. It's hard to tell at a glance what they are testing and whether they are themselves correct. I don't have any ideas for how to make them better. Do you?